Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationWindow.cs

```diff
@@ -0,0 +1,417 @@
+            {
+                MigrateGameObjectHierarchy(parent);
+            }
+            EditorSceneManager.SaveScene(currentScene);
+            EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
+        }
+
+        private void MigrateAllPrefabs()
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+
+            foreach (var path in FindAllAssets(PrefabExtension))
+            {
+                using (var editScope = new EditPrefabAsset(path))
+                {
+                    MigrateGameObjectHierarchy(editScope.root);
+                }
+            }
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+        }
+
+        private void MigrateSelection()
+        {
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+
+            foreach (var migrationObject in migrationObjects)
+            {
+                string path = AssetDatabase.GetAssetPath(migrationObject);
+
+                if (IsAsset(path))
+                {
+                    if (IsExtension(path, PrefabExtension))
+                    {
+                        // Modify and save Prefab Asset
+                        using (var editScope = new EditPrefabAsset(path))
+                        {
+                            GameObject parent = editScope.root;
+                            MigrateGameObjectHierarchy(parent);
+                        }
+                    }
+                    else if (IsExtension(path, SceneExtension))
+                    {
+                        MigrateScene(path);
+                    }
+                }
+                else
+                {
+                    MigrateGameObjectHierarchy((GameObject)migrationObject);
+                }
+            }
+            migrationObjects.Clear();
+        }
+
+        private void MigrateGameObjectHierarchy(GameObject gameObject)
+        {
+            foreach (var child in gameObject.GetComponentsInChildren<Transform>())
+            {
+                if (migrationTypeInstance.CanMigrate(child.gameObject))
+                {
+                    migrationTypeInstance.Migrate(child.gameObject);
+                }
+            }
+        }
+
+        private string[] FindAllAssets(string extension)
+        {
+
+            string[] scenePaths = Directory.GetFiles(Application.dataPath, "*" + extension, SearchOption.AllDirectories);
+
+            for (int i = 0; i < scenePaths.Length; ++i)
+            {
+
+                var scenePath = scenePaths[i];
+                var path = scenePath.Substring(0, scenePath.Length - extension.Length);
+
+                if (!IsAsset(path))
+                {
+                    continue;
+                }
+            }
+            return scenePaths;
+        }
+
+        private bool IsAsset(string path)
+        {
+            return File.Exists(path);
```

Comment:
A little bit confused about this IsAsset check - should we actually be using the AssetDatabase to do some sanity checking? Because otherwise what we're doing in the loop above is getting a list of files that actually exist on disk, and then asking if they exist (we already know that they exist because we used Directory.GetFiles)

Reply:
Checking this